### ISSUE TITLE
nested models not updated properly?

### ISSUE THREAD

**Issue by bluscreen** [emberjs/data#5389]
hi. i think i found a bug in ED 3
given following model hiearchy:

```
antrag.police.vertraege.gefahren.objekte [...]
nested routes: police>vertrag>objekt
```

on an object i set a boolean called bearbeitet from false to true.
after that i save the antrag with everything attached to it

when i go to ember inspector and pick the model from the object controller it will return `true`

but when i call via hierarchy or go thru a store peekrecord or even findrecord its still false which is wrong.

see console output...
x is referencing to $E

```
x.get('police.vertraege.firstObject.gefahr.objekte.firstObject.bearbeitet')
09:51:59.331 false
09:53:58.967 x.get('police.vertraege.firstObject.gefahren.firstObject.objekte.firstObject.bearbeitet')
09:53:58.976 false
09:54:16.009 $E.constructor.modelName
09:54:16.017 "objekt"
09:54:24.344 $E.get('bearbeitet')
09:54:24.348 true
```

after pressing F5 once everything is ok... it works after downgrading to ember 2.18

Update: so far I can only see booleans affected... ?

**Comment by runspired** [comment]
@bluscreen do you have a simple reproduction app or twiddle we could look at?

**Comment by nickschot** [comment]
I've seen the same thing with ember-data 3.1. Seems to have been solved in the latest 3.2 beta when I quickly tried.

**Comment by runspired** [comment]
@nickschot any more info? Still way to fuzzy on details here.

**Comment by nickschot** [comment]
@runspired  I realised that, no time unfortunately :( . I'll try to whip up a reproduction ASAP, not sure if I can make it today.

**Comment by runspired** [comment]
I'm fairly confident that the underlying cause was revealed in #5444 which was also revealed to be a known issue that we have fixed due to recent changes in Ember's array proxy beforeObservers. We will backport the fix to `2.18` `3.0` and `3.1`. It is already available in `3.2`
